test/js/bun/sys/thread-spawn-fail-shim.c

```
@@ -0,0 +1,32 @@
+static int (*real_pthread_create)(pthread_t *, const pthread_attr_t *, void *(*)(void *), void *);
+static const char *marker;
+
+// Without the fix the child aborts. No core file, so CI does not count that as
+// a crash of the test runner; RLIMIT_CORE survives exec.
+__attribute__((constructor)) static void init(void) {
+  struct rlimit rl = {0, 0};
+  setrlimit(RLIMIT_CORE, &rl);
+  marker = getenv("REFUSE_THREADS_WHILE_EXISTS");
+}
+
+int pthread_create(pthread_t *thread, const pthread_attr_t *attr, void *(*start)(void *), void *arg) {
```

Cppcheck (2.21.0)

[style] The function 'pthread_create' is never used.

(unusedFunction)